cpp/src/arrow/filesystem/azurefs_test.cc

```diff
@@ -105,15 +111,41 @@ AzuriteEnv* GetAzuriteEnv() {
   return ::arrow::internal::checked_cast<AzuriteEnv*>(azurite_env);
 }
 
-// Placeholder tests for file structure
+// Placeholder tests
 // TODO: GH-18014 Remove once a proper test is added
-TEST(AzureFileSystem, InitialiseAzurite) {
+TEST(AzureFileSystem, UploadThenDownload) {
+  const std::string containerName = "sample-container";
+  const std::string blobName = "sample-blob.txt";
+  const std::string blobContent = "Hello Azure!";
+
   const std::string& account_name = GetAzuriteEnv()->account_name();
   const std::string& account_key = GetAzuriteEnv()->account_key();
-  EXPECT_EQ(account_name, "devstoreaccount1");
-  EXPECT_EQ(account_key,
-            "Eby8vdM02xNOcqFlqUwJPLlmEtlCDXJ1OUzFT50uSRZ6IFsuFq2UVErCz4I6tq/"
-            "K1SZFPTOtr/KBHBeksoGMGw==");
+
+  auto credential = std::make_shared<Azure::Storage::StorageSharedKeyCredential>(
+      account_name, account_key);
+
+  auto serviceClient = Azure::Storage::Blobs::BlobServiceClient(
+      "http://127.0.0.1:10000/devstoreaccount1", credential);
```

Comment:
Can we use `account_name` instead of `devstoreaccount1`?

Reply:
It looks like changing this actually causes the azurite tests to fail. All the azurite examples I can see online use the same naem `devstoreaccount1`. I suspect there is an assumption built into azurite. 

I can try to investigate further if you feel strongly but for now I'm going to revert to using `devstoreaccount1`.